app/uk/gov/hmrc/hecstubs/models/CompanyAccountingPeriodResponse.scala

```diff
@@ -0,0 +1,152 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hecstubs.models
+
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.hecstubs.models.accountOverviewDetails.ReturnStatus
+import uk.gov.hmrc.hecstubs.models.companyAccountingPeriod.{AccountingPeriod, CompanyAccountingPeriod, CompanyAccountingPeriodRequestParameters, ReturnLookupStatus}
+
+final case class CompanyAccountingPeriodResponse(status: Int, responseBody: JsValue)
+
+object CompanyAccountingPeriodResponse {
+
+  def returnFoundResponse(request: CompanyAccountingPeriodRequestParameters): JsValue =
+    Json.toJson(
+      CompanyAccountingPeriod(
+        request.ctutr,
+        ReturnLookupStatus.Successful,
+        List(
+          AccountingPeriod(
+            request.startDate,
+            request.startDate.plusDays(1L),
+            ReturnStatus.NoReturnFound
+          ),
+          AccountingPeriod(
+            request.endDate.minusDays(1L),
+            request.endDate,
+            ReturnStatus.ReturnFound
+          )
+        )
+      )
+    )
+
+  def noReturnFoundResponse(request: CompanyAccountingPeriodRequestParameters): JsValue =
+    Json.toJson(
+      CompanyAccountingPeriod(
+        request.ctutr,
+        ReturnLookupStatus.Successful,
+        List(
+          AccountingPeriod(
+            request.startDate,
+            request.startDate.plusDays(1L),
+            ReturnStatus.ReturnFound
+          ),
+          AccountingPeriod(
+            request.endDate.minusDays(1L),
+            request.endDate,
+            ReturnStatus.NoReturnFound
+          )
+        )
+      )
+    )
+
+  def noticeToFileIssuedResponse(request: CompanyAccountingPeriodRequestParameters): JsValue =
+    Json.toJson(
+      CompanyAccountingPeriod(
+        request.ctutr,
+        ReturnLookupStatus.Successful,
+        List(
+          AccountingPeriod(
+            request.startDate,
+            request.startDate.plusDays(1L),
+            ReturnStatus.ReturnFound
+          ),
+          AccountingPeriod(
+            request.endDate.minusDays(1L),
+            request.endDate,
+            ReturnStatus.NoticeToFileIssued
+          )
+        )
+      )
+    )
+
+  def noAccountingPeriodsResponse(request: CompanyAccountingPeriodRequestParameters): JsValue =
+    Json.toJson(
+      CompanyAccountingPeriod(
+        request.ctutr,
+        ReturnLookupStatus.NoLiveRecords,
+        List.empty
+      )
+    )
+
+  val error404Response: JsValue =
+    Json.parse(s"""
+       |{
+       |   "code": "NOT_FOUND",
+       |   "reason": " The remote endpoint has indicated that no data can be found for the given identifier."
+       |}
+       |""".stripMargin)
+
+  val error400Response: JsValue =
+    Json.parse(
```

Comment:
Should we include a scenario where we have invalid date format, either start date or end date?

Reply:
yeah, I can add that in - shouldn't be too difficult to add that :+1:

Reply:
updated now - I noticed that I was missing a few test cases in the unit tests as well so I added them in